pkg/operator/encryption/key_controller_encryption_test.go

```diff
@@ -0,0 +1,128 @@
+package encryption
+
+import (
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+func TestEncryptionKeyController(t *testing.T) {
+	scenarios := []struct {
+		name                     string
+		initialSecrets           []runtime.Object
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		validateFunc             func(actions []clientgotesting.Action)
+	}{
+		// scenario 1: assumes a clean slate, that is, there are no previous resources in the system.
+		// It expects that a secret resource with an appropriate key, name and labels will be created.
+		{
+			name: "scenario 1: checks if a secret with AES256 key for core/secret is created",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			validateFunc: func(actions []clientgotesting.Action) {
+				wasSecretValidated := false
+				for _, action := range actions {
+					if action.Matches("create", "secrets") {
+						createAction := action.(clientgotesting.CreateAction)
+						actualSecret := createAction.GetObject().(*corev1.Secret)
+						expectedSecret := createSecretFor("", schema.GroupResource{Group: "", Resource: "secrets"}, 1, actualSecret)
+						if !equality.Semantic.DeepEqual(actualSecret, expectedSecret) {
+							t.Errorf(diff.ObjectDiff(actualSecret, expectedSecret))
+						}
+						wasSecretValidated = true
+						break
+					}
+				}
+				if !wasSecretValidated {
+					t.Errorf("the secret wasn't created and validated")
+				}
+				// TODO: we could actually try to encrypt some data with the key to prove it is valid
```

Comment:
wdyt ?

Reply:
on the other hand, the key contains some random bytes so checking the length of the key should be enough.